### PR TITLE
Cleanup and asserts

### PR DESCRIPTION
**Pull request by Ngalstyan4**

### CODE REVIEW

**Comment by var77** [comment]
By this change, should we use `ldb_invariant` instead of `elog(ERROR)` in all places?

**Comment by Ngalstyan4** [comment]
No, some errors are expected and we can actually trigger them in tests. 
Those should continue using `elog` directly

`ldb_invariant` is for invariants that hold no matter what and we can think of no way to trigger these.
The only reason I still `elog` from `ldb_invariant` is to get some hints on what went wrong, in case we do run into a broken invariant.

**Comment by dqii** [comment]
TODO: Add an invariant check to `ldb_ get_operator_oids`

**Review by var77 — APPROVED** [review]

File: src/hooks/plan_tree_walker.c [hunk 1]
```diff
@@ -55,7 +57,7 @@ bool plan_tree_walker(Plan *plan, bool (*walker_func)(Node *plan, void *context)
         // Join nodes
         case T_Join:
             Join *join = (Join *)plan;
-            if(base_plan_walker(&(join->plan), walker_func, context)) return true;
+            if(base_plan_walker((Plan *)&(join->plan), walker_func, context)) return true;
```
Comment (dqii) [inline src/hooks/plan_tree_walker.c:60]: I don't think this or the one below should be necessary, join->plan is always a plan.

**Review by dqii — APPROVED** [review]

File: src/hooks/plan_tree_walker.c (hunk 1 above)
Comment (Ngalstyan4) [reply src/hooks/plan_tree_walker.c]: You are right. removed it.

**Pull request merged**